### ISSUE TITLE
no-unnecessary-qualifier reports qualifier as being in scope when it is not

### ISSUE THREAD

**Issue by tomblind** [palantir/tslint#2413]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.2.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
//foo.ts
namespace Foo
{
	export class SomeClass
	{
	}
}

//bar.ts
namespace Bar
{
	export var someVar = new Foo.SomeClass();
}
```

with `tslint.json` configuration:

```json
{
	"rules":
	{
		"no-unnecessary-qualifier": true
	}
}
```

#### Actual behavior

tslint reports:
bar.ts[3, 27]: Qualifier is unnecessary since 'Foo' is in scope.

#### Expected behavior

No warning should be emitted, as Foo is not in scope and removing the qualifier will produce code that will not compile.

Note that the two namespaces must be in separate files and the first namespaces declared in those files. Placing a namespace above either Foo or Bar makes the warning go away.

tslint command: tslint -c tslint.json --type-check --project tsconfig.json

tsconfig.json:
```json
{
	"files": [
		"foo.ts",
		"bar.ts"
	]
}
```

**Comment by manu-st** [comment]
I have a similar issue but in a slightly different context and the latest version of tslint (v5.10.0) is reporting the error:

```
export enum FooEnum
{
   FIRST = 0,
   LAST = 1,
   INVALID = 2,
}

export namespace FooEnum
{
   export function isEnumValid (value: FooEnum): boolean
   {
      return value !== FooEnum.INVALID;
   }
}
```

**Comment by zakjan** [comment]
Same invalid error with enum and namespace here.

**Comment by JoshuaKGoldberg** [comment]
The original code (`foo.ts`, `bar.ts`) is no longer an issue, but manu-st's `FooEnum` still does.

**Comment by igabesz** [comment]
Still having this issue with `FooEnum` using tslint@5.18.0

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
